### PR TITLE
Adding Foundry support

### PR DESCRIPTION
**Pull request by RedRise**

### CODE REVIEW

**Comment by antoinedc** [comment]
This is amazing, thanks :)
I will review this week!

File: .gitignore
Comment (antoinedc) [inline .gitignore:1]: This extra line should stay here

File: .vscode/launch.json
Comment (antoinedc) [inline .vscode/launch.json:1]: Can you remove this file & folder, please?

File: bin/index.js [hunk 1]
```diff
@@ -10,9 +10,11 @@ const path = require('path');
 const config = require('../config');
 const inquirer = require('../inquirer');
 const yaml = require('js-yaml');
+const toml = require('toml');
 const TruffleConfig = require('@truffle/config');
 const { parseTrace } = require('../tracer');
 const Api = require('../api');
+const { confirmPasswordReset } = require('firebase/auth');
```
Comment (antoinedc) [inline bin/index.js]: Mmh not sure this is needed, why did you add that? It doesn't seem to be used anywhere either

File: .gitignore
Comment (RedRise) [reply .gitignore:1]: I dont understand why it's marked as a modified file.

File: .vscode/launch.json
Comment (RedRise) [reply .vscode/launch.json:1]: Done

**Comment by RedRise** [comment]
Comments have been integrated

**Comment by antoinedc** [comment]
I'm having troubles testing it. It might be because I've never used Foundry before, but here's what I've done:
- Run an Anvil node
- `foundry init`
- Start the CLI on your branch, it detects the Foundry project successfully
- `forge create src/Counter.sol:Counter --private-key 0x2...`

No changes are detected. It seems like your code is expecting a "broadcast" directory, but I don't have any created.

Any idea?

**Comment by RedRise** [comment]
Hello,

Indeed, it requires to deploy your contract through `forge script`, and not `forge create`.

To make it work, fill the `script/Counter.s.sol` file with the following :

```solidity
 // SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.13;

import "forge-std/Script.sol";
import "../src/Counter.sol";

contract CounterScript is Script {
    function setUp() public {}

    function run() public {
        vm.startBroadcast();
        Counter counter = new Counter();
        vm.stopBroadcast();
    }
}
```
Then, in the command line, run  `forge script --rpc-url http://127.0.0.1:8545 --private-key <insert_your_pk> CounterScript
`

**Comment by antoinedc** [comment]
Aah got it! Everything works perfectly! There is a weird conflict though with your branch. Would you mind fixing it, please? (The correct version is the second block, with "authenticatedUrl")

**Comment by antoinedc** [comment]
I'll merge right after
